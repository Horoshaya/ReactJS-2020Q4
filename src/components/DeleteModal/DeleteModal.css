.title {
  text-transform: uppercase;
  font-weight: 100;
}

.button {
  padding: 10px 15px;
  text-transform: uppercase;
  border: none;
  color: #ffffff;
  background: #f65261;
}
