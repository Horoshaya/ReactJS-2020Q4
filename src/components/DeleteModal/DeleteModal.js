import React from 'react';
import { useDispatch } from 'react-redux';

import Modal from '../Modal/Modal';
import styles from './DeleteModal.css';
import { deletetMovie } from '../../store/actions/actionCreators';

const DeleteModal = ({ triggerModal, id }) => {
  const dispatch = useDispatch();

  const deleteMovieHandle = () => {
    dispatch(deletetMovie(id));
  };
  return (
    <Modal triggerModal={() => triggerModal('delete')}>
      <h2 className={styles.title}>delete movie</h2>
      <p>Are you sure you want to delete this movie?</p>
      <button className={styles.button} onClick={deleteMovieHandle}>
        confirm
      </button>
    </Modal>
  );
};

export default DeleteModal;
// 269149;
