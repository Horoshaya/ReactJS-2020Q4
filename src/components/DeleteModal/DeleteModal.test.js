import React from 'react';
import rerender from 'react-test-renderer';
import DeleteModal from './DeleteModal';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const initialStore = {
  allMoviws: [],
  movies: [
    {
      genres: ['Drama', 'Romance'],
    },
  ],
};

const store = configureStore([])(initialStore);
store.dispatch = jest.fn();

describe('DleteModal', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <Provider store={store}>
        <DeleteModal />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
