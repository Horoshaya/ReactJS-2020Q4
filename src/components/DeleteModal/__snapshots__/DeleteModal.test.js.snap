// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DleteModal then snapshot created 1`] = `
<div>
  <div>
    <span
      onClick={[Function]}
    />
    <h2>
      delete movie
    </h2>
    <p>
      Are you sure you want to delete this movie?
    </p>
    <button
      onClick={[Function]}
    >
      confirm
    </button>
  </div>
  <div />
</div>
`;
